gate-web/src/main/groovy/com/netflix/spinnaker/gate/controllers/AuthController.groovy

```diff
@@ -70,7 +75,11 @@ class AuthController {
    */
   @RequestMapping(value = "/roles/sync", method = RequestMethod.POST)
   void sync() {
-    userRolesSyncer.sync()
+    if (userRolesSyncer) {
```

Comment:
Seems like we are going to have two much different methods of handling users/auth here. I'm not sure we want these code paths to live next to one another in master. Since Gate's behavior is going to change pretty dramatically (compared to the other components, where we are just adding checks), I suggest you make a 'spinnaker:gate-fiat' branch and push the changes to that so you can develop that without worrying about maintaining backwards compatibility in every PR.